tokio/src/runtime/blocking/pool.rs

```diff
@@ -524,17 +524,28 @@ impl Inner {
                 shared = lock_result.0;
                 let timeout_result = lock_result.1;
 
+                // Even if the condvar "timed out", if the pool is entering the
+                // shutdown phase, we want to perform the cleanup logic.
+                if shared.shutdown {
+                    // Work was produced, and we "took" it (by decrementing num_notify).
+                    // This means that num_idle was decremented once for our wakeup.
+                    // But, since we are exiting, we need to "undo" that, as we'll stay idle.
+                    self.metrics.inc_num_idle_threads();
```

Comment:
This comment is confusing to me. Where did we decrement `num_notify`?

I guess what's happening here is that [when waking is up](https://github.com/tokio-rs/tokio/blob/44d11ffcb65303e75e88d2475ec36121c17a954b/tokio/src/runtime/blocking/pool.rs#L441-L448), the counter is decremented, so this increment is there to counteract that.

But that's not what the comment says. Also, what if `wait_timeout` returned due to the timeout triggering, rather than due to us being woken by another thread?